saleor/payment/utils.py

```diff
@@ -189,12 +205,77 @@ def create_payment_information(
         reuse_source=store_source,
         data=additional_data or {},
         graphql_customer_id=graphql_customer_id,
+        refund_data=refund_data,
         _resolve_lines=lambda: create_payment_lines_information(
             payment, manager or get_plugins_manager()
         ),
     )
 
 
+RefundLines = Iterator[Tuple[Any, OrderLine]]
+
+
+def _prepare_refund_lines(
+    order: Order,
+    order_lines_to_refund: List[OrderLineData],
+    fulfillment_lines_to_refund: List[FulfillmentLineData],
+) -> RefundLines:
+    previous_refund_lines: Iterator[Tuple[Any, OrderLine]] = chain.from_iterable(
```

Comment:
Can we re-implement this to something which will not produce so many db queries?
I don't think that you need to iterate overall fulfillments. Can't we call .filter there with status="refunded" there? prefetch_related would be also good.

Also, what about the rest statuses which can also store lines which were refunded? Like returned_and_refunded